framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -0,0 +1,1196 @@
+              tips_tets.insert(neighbor);
+          }
+        }
+
+        // append the missing fine tets (inside the coarse element, away from the node considered)
+        // into the fine elements set for the check on "did it refine the tentative coarse tet
+        // onto the same fine tets"
+        fine_elements.clear();
+        for (const auto & elem : tips_tets)
+          fine_elements.insert(elem);
+        for (const auto & elem : inside_tets)
+          fine_elements.insert(elem);
+
+        // get the vertex of the coarse element from the tip tets
+        for (const auto & tip : tips_tets)
+        {
+          for (const auto & node : tip->node_ref_range())
+          {
+            bool outside = true;
+
+            const auto id = tip->get_node_index(&node);
+            if (!tip->is_vertex(id))
+              continue;
+            for (const auto & tet : inside_tets)
+              if (tet->get_node_index(&node) != libMesh::invalid_uint)
+                outside = false;
+            if (outside)
+            {
+              tentative_coarse_nodes.push_back(&node);
+              // only one tip node per tip tet
+              break;
+            }
+          }
+        }
+
+        std::sort(tentative_coarse_nodes.begin(), tentative_coarse_nodes.end());
+        tentative_coarse_nodes.erase(
+            std::unique(tentative_coarse_nodes.begin(), tentative_coarse_nodes.end()),
+            tentative_coarse_nodes.end());
+
+        // The group of fine elements ended up having less or more than 4 tips, so it's clearly
+        // not forming a coarse tetrahedral
+        if (tentative_coarse_nodes.size() != 4)
+          continue;
+      }
+      else
+      {
+        mooseInfo("Unsupported element type ",
+                  elem_type,
+                  ". Skipping detection for this node and all future nodes near only this "
+                  "element type");
+      }
+
+      // Check the fine element types: if not all the same then it's not uniform AMR
+      for (auto elem : fine_elements)
+        if (elem->type() != elem_type)
+          continue;
+
+      // Check the number of coarse element nodes gathered
+      for (const auto & check_node : tentative_coarse_nodes)
+        if (check_node == nullptr)
+          continue;
+
+      // Form a parent, of the same type as the elements we are trying to combine!
+      std::unique_ptr<Elem> parent;
+      if (elem_type == QUAD4 || elem_type == QUAD8 || elem_type == QUAD9)
+        parent = std::make_unique<Quad4>();
+      else if (elem_type == HEX8 || elem_type == HEX20 || elem_type == HEX27)
+        parent = std::make_unique<Hex8>();
+      else if (elem_type == TRI3 || elem_type == TRI6 || elem_type == TRI7)
+        parent = std::make_unique<Tri3>();
+      else if (elem_type == TET4 || elem_type == TET10 || elem_type == TET14)
+        parent = std::make_unique<Tet4>();
+      else
+        continue;
```

Comment:
was looking for that one smh did not find it

Reply:
done. Hits an assert when trying to refine high order tet and tri

Reply:
Could you give me a stack trace?  I've spent the day (and returned to it in the night...) coding in a panic because you made me realize we don't have enough unit test coverage on TRI7 or TET14 refinement... but TRI6 and TET10 refinement should have been in dbg in CI for a decade, and the new tests I'm adding for the others aren't showing issues.

Reply:
Actually, looking through it after some sleep, we do have TRI7 and TET14 refinement in CI too.  The new unit tests are more focused and added better coverage and caught some bugs in not-yet-tested infinite element methods though, so my panic wasn't a complete waste.  It still leaves me baffled as to what assert you're triggering and I'm failing to, is all.

Reply:
Keep in mind that I am not setting all the nodes to these second order elements. That s probably the problem

Reply:
Ah!  Yeah, that would definitely be an issue.  Weird that it wasn't a problem with tensor product elements too, though.

Would `Elem::build(Elem::first_order_equivalent_type(elem_type))` fix it, then?

Reply:
for sure as that reverts to the previous thing I was doing just a lot shorter in terms of code

Reply:
just did that